flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.catalog.ExternalCatalog;
+import org.apache.flink.table.descriptors.ConnectorDescriptor;
+import org.apache.flink.table.descriptors.TableDescriptor;
+import org.apache.flink.table.functions.ScalarFunction;
+import org.apache.flink.table.sinks.TableSink;
+import org.apache.flink.table.sources.TableSource;
+
+/**
+ * The base class for batch and stream TableEnvironments.
+ *
+ * <p>The TableEnvironment is a central concept of the Table API and SQL integration. It is
+ * responsible for:
+ *
+ * <ul>
+ *     <li>Registering a Table in the internal catalog</li>
+ *     <li>Registering an external catalog</li>
+ *     <li>Executing SQL queries</li>
+ *     <li>Registering a user-defined (scalar, table, or aggregation) function</li>
```

Comment:
Could we update this comment to current state? In the base interface we cannot register table nor aggregation function. I know that this is a temporary thing, but maybe at least link to the implementations that can do that? Sth like: `Registering a user-defined scalar function and table or aggregation function (via concrete implementation)`

Reply:
You are right, we can't register table or aggregate function with the `TableEnvironment` interface. I will correct the comments. Thank you.

Reply:
Just curious, why we can't register table or aggregate function with the TableEnvironment interface now? Sounds like it's only a temporary thing?

Reply:
Just curious, why we can't register table or aggregate function with the TableEnvironment interface now? Sounds like it's only a temporary thing?

Reply:
We still can register functions, just not on the unified Environment. You can still register them on `Stream/BatchTableEnvironment`, which are the only entry points. We will allow registering aggregate & table functions on `TableEnvironment` once we rework type system.